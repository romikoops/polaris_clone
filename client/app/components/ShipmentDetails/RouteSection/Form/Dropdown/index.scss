:local {
  .route_dropdown {
    margin: 10px auto;
    width: 100%;
  }
}