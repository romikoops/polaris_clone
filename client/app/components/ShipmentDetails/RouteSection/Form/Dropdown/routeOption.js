export default function routeOption (route) {
  return {
    label: route.country ? `${route.nexusName}, ${route.country}` : route.nexusName,
    value: {
      id: route.nexusId,
      latitude: route.latitude,
      longitude: route.longitude,
      name: route.nexusName,
      country: route.country
    }
  }
}
