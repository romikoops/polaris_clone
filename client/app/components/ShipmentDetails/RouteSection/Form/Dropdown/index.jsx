import React from 'react'
import { withNamespaces } from 'react-i18next'
import FormsySelect from '../../../../Formsy/Select'
import styles from './index.scss'
import routeOption from './routeOption'

const customStyles = `
  .Select-control {
    background-color: #F9F9F9;
    box-shadow: 0 2px 3px 0 rgba(237, 234, 234, 0.5);
    border: 1px solid #f2f2f2 !important;
  }
  .Select-menu-outer {
    box-shadow: 0 2px 3px 0 rgba(237, 234, 234, 0.5);
    border: 1px solid #f2f2f2;
  }
  .Select-value {
    background-color: #F9F9F9;
    border: 1px solid #f2f2f2;
  }
  .Select-option {
    background-color: #f9f9f9;
  }
`

function getOptions (targets) {
  const labels = []
  const options = []

  targets.forEach((target) => {
    const option = routeOption(target)
    if (labels.includes(option.label)) return

    labels.push(option.label)
    options.push(option)
  })

  return options.sort((a, b) => (a.label > b.label ? 1 : -1))
}

function Dropdown ({
  target,
  availableTargets,
  formData,
  onDropdownSelect,
  t
}) {
  return (
    <div className={`dropdown ${styles.route_dropdown}`}>
      <FormsySelect
        name={`${target}-nexus`}
        className={styles.select}
        value={formData.nexusId ? routeOption(formData) : null}
        placeholder={t(`shipment:${target}`)}
        options={getOptions(availableTargets)}
        onChange={option => onDropdownSelect(target, option)}
        customStyles={customStyles}
        validationErrors={{ isDefaultRequiredValue: t('errors:notBlank') }}
        required
      />
    </div>
  )
}

export default withNamespaces(['shipment', 'errors'])(Dropdown)
