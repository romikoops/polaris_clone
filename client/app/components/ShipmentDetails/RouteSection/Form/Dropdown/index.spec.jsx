import * as React from 'react'
import { shallow } from 'enzyme'
import Dropdown from '.'

const availableTargets = [
  {
    stopId: 4673,
    hubId: 3023,
    hubName: 'Gothenburg Port',
    nexusId: 597,
    nexusName: 'Gothenburg',
    latitude: 57.694253,
    longitude: 11.854048,
    country: 'SE',
    truckTypes: [
      'default'
    ]
  },
  {
    stopId: 33903,
    hubId: 5961,
    hubName: 'Southampton Port',
    nexusId: 3282,
    nexusName: 'Southampton',
    latitude: 50.9097004,
    longitude: -1.4043509,
    country: 'GB',
    truckTypes: [
      'default'
    ]
  }
]

const propsBase = {
  target: 'TARGET',
  availableTargets,
  formData: {},
  onDropdownSelect: x => x
}

test('with empty props', () => {
  expect(() => shallow(<Dropdown />)).toThrow()
})

test('happy path', () => {
  expect(shallow(<Dropdown {...propsBase} />)).toMatchSnapshot()
})
