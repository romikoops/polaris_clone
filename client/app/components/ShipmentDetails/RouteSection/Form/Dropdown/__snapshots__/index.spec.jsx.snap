// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`happy path 1`] = `
<div
  className="dropdown route_dropdown"
>
  <Formsy(FormsySelect)
    className="select"
    customStyles="
  .Select-control {
    background-color: #F9F9F9;
    box-shadow: 0 2px 3px 0 rgba(237, 234, 234, 0.5);
    border: 1px solid #f2f2f2 !important;
  }
  .Select-menu-outer {
    box-shadow: 0 2px 3px 0 rgba(237, 234, 234, 0.5);
    border: 1px solid #f2f2f2;
  }
  .Select-value {
    background-color: #F9F9F9;
    border: 1px solid #f2f2f2;
  }
  .Select-option {
    background-color: #f9f9f9;
  }
"
    innerRef={[Function]}
    name="TARGET-nexus"
    onChange={[Function]}
    options={
      Array [
        Object {
          "label": "Gothenburg",
          "value": Object {
            "country": "SE",
            "id": 597,
            "latitude": 57.694253,
            "longitude": 11.854048,
            "name": "Gothenburg",
          },
        },
        Object {
          "label": "Southampton",
          "value": Object {
            "country": "GB",
            "id": 3282,
            "latitude": 50.9097004,
            "longitude": -1.4043509,
            "name": "Southampton",
          },
        },
      ]
    }
    placeholder="NO_TRANSLATION | key \\"shipment:TARGET\\""
    required={true}
    validationError=""
    validationErrors={
      Object {
        "isDefaultRequiredValue": "Must not be blank",
      }
    }
    validations={null}
    value={null}
  />
</div>
`;
